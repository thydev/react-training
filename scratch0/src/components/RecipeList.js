import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

const RecipeList = ({
  recipes, style, favorites, ...props
}) => (
  <ul style={style} className="list-reset">
    {recipes.map(recipe => (
      <RecipeListItem
        key={recipe.id}
        recipe={recipe}
        favorited={favorites.includes(recipe.id)}
        {...props}
      />
    ))}
  </ul>
);
RecipeList.propTypes = {
  style: PropTypes.object,
  recipes: PropTypes.array,
  favorites: PropTypes.array,
  onClick: PropTypes.func,
  onFavorited: PropTypes.func,
};

RecipeList.defaultProps = {
  recipes: [],
  favorites: [],
};
export default RecipeList;
