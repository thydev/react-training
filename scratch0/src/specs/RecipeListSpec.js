import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const testRecipes = [
  {
    id: 1,
    name: 'Test recipe1',
    category: 'Test category1',
  },
  {
    id: 2,
    name: 'Test recipe2',
    category: 'Test category2',
  },
];

describe('<RecipeList />', () => {
  test('Should not break when no recipes passed', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render recipes correctly', () => {
    const component = renderer.create(<RecipeList recipes={testRecipes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorite state correctly', () => {
    const component = renderer.create(<RecipeList recipe={testRecipes} favorites={[1]} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
