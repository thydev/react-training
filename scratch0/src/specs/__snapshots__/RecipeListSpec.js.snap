// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Should not break when no recipes passed 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList /> Should render favorite state correctly 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList /> Should render recipes correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ⬜️
    </span>
    <span>
      Test recipe1
    </span>
    <span>
      Test category1
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ⬜️
    </span>
    <span>
      Test recipe2
    </span>
    <span>
      Test category2
    </span>
  </li>
</ul>
`;
